src/Content/Version.php

```diff
@@ -109,77 +152,70 @@ public function model(): ModelWithContent
+		Language|string $language = 'default'
 	): int|null {
-		$this->ensure($language);
-		return $this->model->storage()->modified($this->id, $language);
+		if ($this->exists($language) === true) {
+			return $this->model->storage()->modified($this->id, $this->language($language));
+		}
+
+		return null;
 	}
 
 	/**
 	 * Moves the version to a new language and/or version
+	 *
+	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
 	 */
-	public function move(string $fromLanguage, VersionId $toVersionId, string $toLanguage): void
-	{
+	public function move(
+		Language|string $fromLanguage,
+		VersionId $toVersionId,
+		Language|string $toLanguage
+	): void {
 		$this->ensure($fromLanguage);
-		$this->model->storage()->move($this->id, $fromLanguage, $toVersionId, $toLanguage);
+		$this->model->storage()->move(
+			fromVersionId: $this->id,
+			fromLanguage: $this->language($fromLanguage),
+			toVersionId: $toVersionId,
+			toLanguage: $this->language($toLanguage)
+		);
 	}
 
 	/**
 	 * Returns the stored content fields
 	 *
-	 * @param string $lang Code `'default'` in a single-lang installation
 	 * @return array<string, string>
+	 *
+	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
 	 */
-	public function read(string $language = 'default'): array
+	public function read(Language|string $language = 'default'): array
 	{
 		$this->ensure($language);
-		return $this->model->storage()->read($this->id, $language);
+		return $this->model->storage()->read($this->id, $this->language($language));
 	}
 
 	/**
 	 * Updates the modification timestamp of an existing version
 	 *
-	 * @param string $lang Code `'default'` in a single-lang installation
-	 *
 	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
 	 */
-	public function touch(string|null $language = null): void
+	public function touch(Language|string $language = 'default'): void
 	{
-		// touch all languages
-		if ($language === null) {
-			foreach ($this->model->kirby()->languages() as $language) {
-				$this->touch($language->code());
-			}
-		}
-
-		// make sure the version exists
 		$this->ensure($language);
-
-		// touch the default language in single-language mode
-		if ($this->model->kirby()->multilang() === false) {
-			$this->model->storage()->touch($this->id, 'default');
-			return;
-		}
-
-		// touch a single language
-		$this->model->storage()->touch($this->id, $language);
+		$this->model->storage()->touch($this->id, $this->language($language));
 	}
 
 	/**
 	 * Updates the content fields of an existing version
 	 *
 	 * @param array<string, string> $fields Content fields
-	 * @param string $lang Code `'default'` in a single-lang installation
 	 *
 	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
 	 */
-	public function update(array $fields, string $language = 'default'): void
+	public function update(array $fields, Language|string $language = 'default'): void
```

Comment:
`Suggestion`: For consistency with the storage classes, I wonder whether we should flip the argument order in `create()` and `update()`. Would also make the language the first argument for all version methods.

Reply:
Chiming in here from the side, but I'd have required parameters before optional ones. I think newer PHP even requires it.

Reply:
Right, that's a good point.

Reply:
I chose consistency with the ModelWithContent::update and ModelWithContent::save here instead. It's a bit weird because it's a different order than in the storage handlers, but I think it's consistent in both places now and Version is acting some sort of a translator.